.github/workflows/integration.yml

```diff
@@ -7,180 +9,243 @@ on:
-    - name: Download bundle itowns
-      uses: actions/download-artifact@v2
-      with:
-        name: dist-itowns
-    - name: Download bundle potree
-      uses: dawidd6/action-download-artifact@v2
-      with:
-        workflow: intergration.yml
-        repo: LIDAR-HD-IGN/potree
-        branch: itowns_potree
-        name: dist-potree
-    # Copy files for deployment
-    - name: build site
-      run: |
-        mkdir -p itowns/dist
-        mkdir -p itowns/potree/build
-        mkdir -p itowns/potree/libs
-        cp -R dist/*.js itowns/dist/
-        cp -R examples itowns/
-        cp -R buildDocs itowns/docs
-        cp -R build itowns/potree/
-        cp -R libs itowns/potree/
-
-    # Deploy to itowns.github.io
-    - name: Deploy to itowns.github.io
-      uses: peaceiris/actions-gh-pages@v3
-      with:
-        deploy_key: ${{ secrets.ACTIONS_DEPLOY_KEY }}
-        external_repository: iTowns/itowns.github.io
-        publish_dir: ./itowns
-        destination_dir: ./itowns
-        publish_branch: master
-        enable_jekyll: true
-
-  # Create github release
+
+      # Download artifact from build
+      - name: Download itowns bundle
+        uses: actions/download-artifact@v2
+        with:
+          name: dist-itowns
+
+      # Deploy potree repo artifact
+      # THIS IS COMMENTED AT THE MOMENT AS POTREE ARTIFACT IS NOT USED YET IN EXAMPLES.
+      # TODO : this should be done in potree integration script
+      # - name: Download potree bundle
+      #   uses: dawidd6/action-download-artifact@v2
+      #   with:
+      #     # TODO : rename potree file...
+      #     workflow: intergration.yml
+      #     repo: LIDAR-HD-IGN/potree
+      #     branch: itowns_potree
+      #     name: dist-potree
+
+      # Copy files for deployment
+      - name: build site
+        run: |
+          mkdir -p itowns/dist
+          mkdir -p itowns/potree/build
+          mkdir -p itowns/potree/libs
+          cp -R dist/*.js itowns/dist/
+          cp -R examples itowns/
+          cp -R buildDocs itowns/docs
+          # cp -R build itowns/potree/
+          # cp -R libs itowns/potree/
+
+      - name: add dev bundle if release
+        if: ${{ startsWith(github.event.head_commit.message, 'release v' ) }}
+        run: |
+          cp -R itowns dev
+          mv dev itowns/
+
+      # Deploy to itowns.github.io LTS
+      - name: Deploy LTS to itowns.github.io
+        if: ${{ startsWith(github.event.head_commit.message, 'release v' ) }}
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          deploy_key: ${{ secrets.ACTIONS_DEPLOY_KEY }}
+          external_repository: iTowns/itowns.github.io
+          publish_dir: ./itowns
+          destination_dir: ./itowns
+          publish_branch: master
+          enable_jekyll: true
+
+      # Deploy to itowns.github.io Dev
+      - name: Deploy Dev to itowns.github.io
+        if: ${{ !startsWith(github.event.head_commit.message, 'release v' ) }}
```

Comment:
I had to do this to prevent having two pushes on iTowns/itowns.github.io when making a release. This would cause an error on the second push, since apparently the same ssh connection can't be used simultaneously for multiple transfers. I tried to setup two separate ssh connections with iTowns/itowns.github.io : one for pushing releases and the other one for pushing next build bundles, but this also failed.
What I do then is preventing the deployment of @next version when deploying a release (thanks to this `if` statement), and copying the release bundles in the dev folder on iTowns/itowns.github.io (on lines 153 to 157 of this script).

Reply:
Ah ok! thanks for the explaination! Maybe we should add this explanation as a comment ?

Reply:
Yep, I added comments here and before line 153.